Sources/Main.cpp

```diff
@@ -1174,51 +1139,101 @@ void __fastcall TX584Form::PasteItemClick(TObject *Sender)
                 CPU.Format(Code[index + i], str);
                 CodeListView->Items->Item[index + i]->SubItems->Strings[1] = str;
                 CodeListView->Items->Item[index + i]->SubItems->Strings[2] = CMClipboard[i];
+                CodeListView->Items->Item[index + i]->Selected = true;
             }
-        SelCount = ClipboardSize;
-        if (index + SelCount > MAX_ADDR)
-            SelCount = MAX_ADDR - index;
+
         CodeListView->Repaint();
         SetModifyFlag(true);
     } else
         PostMessageW(ActiveControl->Handle, WM_PASTE, 0, 0);
 }
 //---------------------------------------------------------------------------
 
+void TX584Form::ClearSelectedItems()
+{
+    TListItem *Item;
+
+    Item = CodeListView->Selected;
+    TItemStates selected = TItemStates() << isSelected;
+    for (; Item; Item = CodeListView->GetNextItem(Item, sdBelow, selected)) {
+        int Index = Item->Index;
+        Code[Index] = NOP;
+        Item->SubItems->Strings[1] = NOP_TEXT;
+        Item->SubItems->Strings[2] = L"";
+    }
+}
+//---------------------------------------------------------------------------
+
+void TX584Form::RemoveSelectedItems()
+{
+    bool NewItems[MAX_ADDR] = {0};
+
+    for (int i = 0; i < MAX_ADDR; i++) {
+        NewItems[i] = true;
+    }
+
+    TListItem *Item;
+    Item = CodeListView->Selected;
+
+    TItemStates selected = TItemStates() << isSelected;
+
+    for (; Item; Item = CodeListView->GetNextItem(Item, sdBelow, selected)) {
+        int Index = Item->Index;
+        NewItems[Index] = false;
+    }
+
+    int NewIndex = 0;
+    int OldIndex = 0;
+
+    for (; OldIndex < MAX_ADDR && NewIndex < MAX_ADDR; OldIndex++) {
```

Comment:
`OldIndex` will be always greater than or equal to `NewIndex` so comparing `NewIndex` with `MAX_ADDR` is not necessary.

```cpp
for (; OldIndex < MAX_ADDR; OldIndex++) {
```